adapters/sidekiq/lib/opentelemetry/adapters/sidekiq/middlewares/client/tracer_middleware.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Sidekiq
+      module Middlewares
+        module Client
+          class TracerMiddleware
+            def call(worker_class, job, _queue, _redis_pool)
+              tracer.in_span(
+                worker_class,
```

Comment:
Is `worker_class` a string? If not, we should convert it here.

Reply:
Good catch, when `sidekiq/testing` is used it comes in as a string which is why I hadn't explicitly `.to_s` the worker_class.  However in a non test environment it is sending in the class constant.

The `job` hash contains job class in string form so I'll pull from there instead.

Reply:
When double checking this I noticed that the job hash contains different information when using the [ActiveJob setup](https://github.com/mperham/sidekiq/wiki/Active-Job#active-job-setup)

With ActiveJob setup
```rb
{
  "retry"=>0,
  "queue"=>"default",
  "class"=>"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper",
  "wrapped"=>SimpleJob,
  "args"=> [
    {
      "job_class"=>"SimpleJob",
      "job_id"=>"8a34d851-ab9d-484b-ada4-320e8e591911",
      "provider_job_id"=>nil,
      "queue_name"=>"default",
      "priority"=>nil,
      "arguments"=>[],
      "executions"=>0,
      "exception_executions"=>{},
      "locale"=>"en",
      "timezone"=>"UTC",
      "enqueued_at"=>"2020-03-27T16:06:24Z"
    }
  ],
  "jid"=>"fcc6967ada593724d55e15a9",
  "created_at"=>1585325184.135893
}
```

Without ActiveJob setup
```rb
{
  "retry"=>0,
  "queue"=>"default",
  "class"=>"SimpleJob",
  "args"=>[],
  "jid"=>"d9bf4951a65e81456a1c7990",
  "created_at"=>1585325499.9481258
}
```

My question regarding the above is, should it be a configuration option to declare whether or not you are using sidekiq with the active job queue adapter?  

Or should this adapter try and handle capturing the job class implicitly doing something like
```rb
job_class = job['wrapped'] || job['class']
```

Not sure what's more in the spirit of this gem.

Reply:
It's nice if we can auto-extract. My recollection is that this is a royal PITA. We have some weirdness in https://github.com/Shopify/shopify-tracing/blob/master/lib/shopify/tracing/adapters/sidekiq_adapter.rb and https://github.com/Shopify/shopify-tracing/blob/master/lib/shopify/tracing/adapters/resque_like_adapter.rb to try to deal with it, and I believe that broke at least one dependent service.

Maybe we can try the auto-extraction, and if we find breakage when people try to use this in production, we can add the config option.